src/main/kotlin/net/maizegenetics/phgv2/pathing/AlignmentUtils.kt

```diff
@@ -0,0 +1,292 @@
+package net.maizegenetics.phgv2.pathing
+
+import it.unimi.dsi.fastutil.longs.Long2LongOpenHashMap
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.channels.SendChannel
+import net.maizegenetics.phgv2.api.HaplotypeGraph
+import net.maizegenetics.phgv2.api.ReferenceRange
+import net.maizegenetics.phgv2.utils.getBufferedWriter
+import net.maizegenetics.phgv2.utils.getBufferedReader
+import org.apache.logging.log4j.LogManager
+import java.io.*
+import java.util.*
+import kotlin.math.ceil
+import kotlin.math.min
+
+/**
+ * data class to hold the KmerMap information.  From the PHGv1 source code:
+ */
+//data class KmerMapData(val haplotypeListId: Int, val rangeToBitSetMap: Map<Int, BitSet>, val kmerHashToLongMap: Long2LongOpenHashMap)
+data class KmerMapData(val rangeToBitSetMap: Map<ReferenceRange, BitSet>, val kmerHashToLongMap: Long2LongOpenHashMap)
+
+data class KeyFileData(val sampleName: String, val file1: String, val file2: String = "")
+
+private val myLogger = LogManager.getLogger("net.maizegenetics.phgv2.utils.AlignmentUtils")
+
+
+fun alignReadsToHaplotypes(hvcfDir: String, kmerIndexFile:String, keyFileRecords:List<KeyFileData>, paired:Boolean, outputDir:String) {
```

Comment:
What data is in the kmerIndexFile?  Can you add a comment header to this function?

Reply:
The kmerIndex file is coming from the previous step.  I will add in a comment describing this.  Again this code is not complete yet and the purpose of this PR is to get the IO in for kmerIndex and readMappings.